proxy/proxy.go

```diff
@@ -0,0 +1,30 @@
+package proxy
+
+import (
+	"net"
+	"net/http"
+	"time"
+)
+
+// MakeHTTPClient makes a HTTP client with good defaults for timeouts.
+func MakeHTTPClient(timeout *time.Duration) http.Client {
+	if timeout != nil {
+		return http.Client{
+			Timeout: 3 * time.Second,
+			Transport: &http.Transport{
+				Proxy: http.ProxyFromEnvironment,
+				DialContext: (&net.Dialer{
+					Timeout: 3 * time.Second,
+					// KeepAlive: 0,
+				}).DialContext,
+				// MaxIdleConns:          1,
+				// DisableKeepAlives:     true,
```

Comment:
Prefer not to see commented out code being be pushed

Reply:
Right - I'm not certain about the perfect options yet so may leave in while we test.